### ISSUE TITLE
allow Waffle sizes to optionally populate based on tags

### ISSUE THREAD

**Issue by ashumz** [waffleio/waffle.io#1720]

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/695270112)

"Since story points in the past are in GitHub labels, is there any API to programmatically assign a Waffle.io story point to a GitHub issue, as well as retrieving the number of story points?

Or is there any plan to integrate with GitHub labels anytime?"

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/700887453)

"Hey, we like the idea of the new task sizing feature, but we're reluctant to use it because the data isn't visible in Github. Currently we indicate sizing with Github tags like "size-small", "size-med", etc. We'd like it if Waffle sizing could optionally store tags."

**Comment by ashumz** [comment]
[+1](https://app.intercom.io/a/apps/fb774a034de340cde33822ae0c071ad60efe899f/inbox/unassigned/conversations/696192179)

"As I sent in the screenshot, I love your product cause it's sync with github, so I love your feature to calculate size value, but, I expected to get this on github too."

**Comment by superkhau** [comment]
+1 Sizing feature should allow a tag to automatically be created in GitHub

**Comment by woisme** [comment]
Love for the size to follow the issue not be restricted to the board

**Comment by dankohn** [comment]
+1. Sizes are a great addition to waffle, but it's the first data that doesn't rely on the Github repository. We want to be able to filter on size in the Github UI. Also, we would like each change to size to be logged to the Github issue, which we would get for free since Github records when (and who) changes labels.

**Comment by mohierf** [comment]
+1 for using GitHub labels for size card information.

**Comment by davidrans** [comment]
+1 This is the only reason we're currently debating using your software.

**Comment by kwiens** [comment]
+1 This would make a big difference for our team. It would allow us to build scripts on top of Waffle and let us make our own throughput charts.

**Comment by JWGmeligMeyling** [comment]
:+1:

**Comment by ehausen** [comment]
+1
Absolutely! as @dankohn stated this the one piece of data that is not stored on Github and can be a dealbraker. The columns labels can be defined. Why not do the same for size? I think this app should follow the principle of always storing everything in Github also, so that other people joining in can just follow and work on github if they liked to

**Comment by hlm** [comment]
+1

**Comment by ruuts** [comment]
:+1:

**Comment by erichexter** [comment]
:+1:

**Comment by kwiens** [comment]
+1

**Comment by lakruzz** [comment]
@ehausen:
> I think this app should follow the principle of always storing everything in Github also, so that other people joining in can just follow and work on github if they liked to

:+1: I second that 100%. 

I'm sorry to see that this issue is a year old, and has not had any comments for the past 3 months. It should get a lot of focus! This is indeed a deal breaker. We have implemented our own labels in GitHub and we would not - for a second - consider utilizing data, that wasn't accessible through an API to GitHub (we use [ghi](https://github.com/stephencelis/ghi)).

Waffles, is there anything we can do to help? Say the word!

**Comment by ashumz** [comment]
Hey all, 

Just an update on this -- we're not ignoring you. :) We have two things in progress to address a lot of the pain around this. 

First, we're releasing a public API in the next few months that will allow you to grab size from the cards, which for many of you will allow you to build the metrics you want. Additionally, we're working on a milestone burn down graph to help with the pain around metrics (will work across multirepo boards, too. :)) 

Drilling down to the specific request here -- for some users, adding a label for every size they add across any given repo would be label overload, whereas many of you would find this useful. For now, the best workaround would be to create the labels in GitHub for the sizes you want in order to keep everything in GitHub.  We might consider the inverse functionality (if you had a label on your board which has a size that we recognize/support, we could show it in the 'size' field and perhaps hide the label on the card, if there's value to that? Feedback welcome. :)) 

(Also, eventually we plan to have label and milestone management for multirepo boards - where you'd be able to create a label across all sources connected to your board. That's a ways out on our roadmap, but would be an easy way for you to manage sizing across labels across multiple repos.) 

We don't have anything planned around this right now, but any feedback you have is welcome and reviewed by the team. :)

Ashley

**Comment by kwiens** [comment]
>  (if you had a label on your board which has a size that we recognize/support, we could show it in the 'size' field and perhaps hide the label on the card, if there's value to that? Feedback welcome. :))

Yes, this! This is precisely what we need. We already use numeric labels in github and then manually sync them to waffle. 

![image](https://cloud.githubusercontent.com/assets/129007/12991839/06b06d44-d0c6-11e5-89c9-f239beed52ec.png)

This would be simple to implement:

1. Set the waffle size to match the numeric label on the issue, if there is one
2. Hide numeric labels from waffle cards

Thanks for thinking about this!

**Comment by ruuts** [comment]
@ashumz what @kwiens is saying sounds perfect!

**Comment by JWGmeligMeyling** [comment]
@kwiens But this would be a one-way solution then right? What would the expected behaviour be if you now change the size in Waffle? Would it remove the old tag and add the new one? Would it create such tag if it was not existing? And how would you differentiate users that want to use this, and ones that do not? Or do you think that this should be the new standard behaviour?

**Comment by kwiens** [comment]
Ideally, when you change the size in waffle it would change the label, like it does when you move cards between columns. Yes, it would add a new tag.

This probably needs to be a setting to avoid imposing this workflow on everyone, although I think most people using size scopes would like a way to see those scopes in Github. Zenhub and Codetree already have a mechanism for this.

**Comment by webcodecs** [comment]
+1

**Comment by lakruzz** [comment]
I Imagine what @kwiens describes simply could be configurable in the waffle boards - mapping labels to sizes - like we're already mapping labels to columns:

![image](https://cloud.githubusercontent.com/assets/155492/13792334/a5128022-eaf1-11e5-87f1-6cf1d3f988db.png)

**Comment by jeanfw** [comment]
+1, this is one of the things that took us time as we moved from Codetree to trying Waffle and had to manually set size on our issues based on our Codetree `size: N` labels

@ashumz 2-way would be better of course, also because seeing label changes on size change helps us keep track of an issue's lifecycle. 

Would be great to configure the format for "N" points as part of onboarding if it turns out there are too many different conventions for size labels:
```N``` (cf. @kwiens)
 ```size: N``` (cf. Codetree)
etc. 
Leaving the format blank would not read nor write size labels, for people who feel it would cause label overload.

**Comment by kwiens** [comment]
I'm good with `size: N`. Let's standardize the industry!

**Comment by zspencer** [comment]
I would also love if whenever a story gets pointed a label for that point value gets attached in Github. 

As it stands now, if I want to do any kind of metricing I do it on a 'issue quantity' basis; which is perfectly fine for me but our product manager really wants points.

**Comment by nickolasclarke** [comment]
👍 Would love to have this, I want the repo to be the single source of truth for everything, while leveraging waffle.io for its superior card management. Best of both worlds!

**Comment by chrowe** [comment]
We are running into this issue as well. We are currently managing sizes in both labels and waffle. We can't keep doing this though.

I think a good option would be to add a checkbox in the settings to "Use Github labels for sizes"  and then tell you the format to use for your labels. This would be quicker than having to map each label one by one.

A future feature might be to allow a configurable syntax for th label names.

**Comment by sirkirby** [comment]
:thumbsup: for this one as well. Really hoping to see this feature in Waffle soon.

**Comment by ToriM** [comment]
👍

**Comment by rmateu-aeropost** [comment]
+1 on `size: N` please.

**Comment by lakruzz** [comment]
@ashumz I just sent you a mail - offering [our](www.prolike.io) help to implement this (again). It's not going to cost you a dime - we'll volunteer,  - We just need to work together.

Solving this issue would make the world a better place!

Let me know if my mail reached you and if you are still the one within CA to hold the leach on this one?

**Comment by rmateu-aeropost** [comment]
@marydavis w00t, great news seeing movement on size sync. We're running a _pilot_ with one of our teams on waffle, and it's going very well – the double work of having to go to the issue on github to size the task has been the biggest feedback.

**Comment by marydavis** [comment]
Sad news, I’m going to have to move this issue back to ‘planned’. I realized this is a bit more involved than originally thought. This would be much easier to implement on new projects but the optional migration of current project sizes makes it bit more complicated. As we recently lost one of our devs, we have to be more conscious of the work we pull. So for now, this will be moved back to planned until we can break it down into smaller pieces and have a better idea of all the work needed to implement this well.
